test/hotspot/jtreg/compiler/vectorapi/VectorCommutativeOperSharingTest.java

```diff
@@ -0,0 +1,413 @@
+    static {
+        ia = new int[LENGTH];
+        ib = new int[LENGTH];
+        ir1 = new int[LENGTH];
+        ir2 = new int[LENGTH];
+        ir3 = new int[LENGTH];
+        ir4 = new int[LENGTH];
+
+        for (int i = 0; i < LENGTH; i++) {
+            ia[i] = RD.nextInt(25);
+            ib[i] = RD.nextInt(25);
+        }
+    }
+
+    public static void main(String[] args) {
+        TestFramework.runWithFlags("--add-modules=jdk.incubator.vector");
+    }
+
+    @Test
+    @IR(counts = {IRNode.ADD_VI, IRNode.VECTOR_SIZE_ANY, " 2 ",
+                  IRNode.MUL_VI, IRNode.VECTOR_SIZE_ANY, " 2 ",
+                  IRNode.MAX_VI, IRNode.VECTOR_SIZE_ANY, " 2 ",
+                  IRNode.MIN_VI, IRNode.VECTOR_SIZE_ANY, " 2 "})
+    public void testVectorIRSharing1(int index) {
+        IntVector vec1 = IntVector.fromArray(I_SPECIES, ia, index);
+        IntVector vec2 = IntVector.fromArray(I_SPECIES, ib, index);
+        vec1.lanewise(VectorOperators.ADD, vec2)
+            .lanewise(VectorOperators.ADD, vec2.lanewise(VectorOperators.ADD, vec1))
+            .intoArray(ir1, index);
+        vec1.lanewise(VectorOperators.MUL, vec2)
+            .lanewise(VectorOperators.MUL, vec2.lanewise(VectorOperators.MUL, vec1))
+            .intoArray(ir2, index);
+        vec1.lanewise(VectorOperators.MAX, vec2)
+            .lanewise(VectorOperators.MAX, vec2.lanewise(VectorOperators.MAX, vec1))
+            .intoArray(ir3, index);
+        vec1.lanewise(VectorOperators.MIN, vec2)
+            .lanewise(VectorOperators.MIN, vec2.lanewise(VectorOperators.MIN, vec1))
+            .intoArray(ir4, index);
+    }
+
+    @Run(test = "testVectorIRSharing1")
+    public void testVectorIRSharingDriver1() {
+        for (int i = 0; i < I_SPECIES.loopBound(LENGTH); i += I_SPECIES.length()) {
+            testVectorIRSharing1(i);
+        }
+    }
+
+    @Test
+    @IR(counts = {IRNode.XOR_VI, IRNode.VECTOR_SIZE_ANY, " 0 ",
+                  IRNode.OR_VI, IRNode.VECTOR_SIZE_ANY, " 1 ",
+                  IRNode.AND_VI, IRNode.VECTOR_SIZE_ANY, " 1 "})
+    public void testVectorIRSharing2(int index) {
+        IntVector vec1 = IntVector.fromArray(I_SPECIES, ia, index);
+        IntVector vec2 = IntVector.fromArray(I_SPECIES, ib, index);
+        vec1.lanewise(VectorOperators.XOR, vec2)
+            .lanewise(VectorOperators.XOR, vec2.lanewise(VectorOperators.XOR, vec1))
+            .intoArray(ir1, index);
+        vec1.lanewise(VectorOperators.AND, vec2)
+            .lanewise(VectorOperators.AND, vec2.lanewise(VectorOperators.AND, vec1))
+            .intoArray(ir2, index);
+        vec1.lanewise(VectorOperators.OR, vec2)
+            .lanewise(VectorOperators.OR, vec2.lanewise(VectorOperators.OR, vec1))
+            .intoArray(ir3, index);
+    }
+
+    @Run(test = "testVectorIRSharing2")
+    public void testVectorIRSharingDriver2() {
+        for (int i = 0; i < I_SPECIES.loopBound(LENGTH); i += I_SPECIES.length()) {
+            testVectorIRSharing2(i);
```

Comment:
Adding some result verification would be nice. Because now you only check that you get the right number of instructions - but that does not mean that they are connected correctly. Result verification could help here.

See:
https://github.com/openjdk/jdk/blob/master/test/hotspot/jtreg/compiler/lib/verify/Verify.java
https://github.com/openjdk/jdk/blob/master/test/hotspot/jtreg/testlibrary_tests/verify/examples/TestVerifyInCheckMethod.java

Also, for random value generation we hope to use this in the future:
https://github.com/openjdk/jdk/blob/master/test/hotspot/jtreg/compiler/lib/generators/Generators.java

Reply:
You don't have to use Generators here, and it is up to you if you want to use `Verify.checkEQ` or do your own comparisons - use what is more convenient for you for now.